addon/helpers/multi-select.js

```diff
@@ -15,7 +15,7 @@ export default Helper.extend({
   },
   setupRecompute (selectedItems, property) {
     if (this.teardown) this.teardown()
-    var path = '@each.id'
+    var path = '[]'
```

Comment:
Even though this might fix the error that was manifesting in your consuming app is this change aligned with providing the same behavior this add-on was previously?  Previously a specific property on each array element was being watched for changes and now we're reacting to whether items are added to or removed from the array, or if the array property is set to a different array.

This is definitely a difference in behavior.  This doesn't change any other behavior of this add-on in an undesired way?  If the change is not undesired, meaning we're aware that this changes the behavior of this add-on then this should not be a PATCH release and instead be a MAJOR.

Reply:
And no tests failed?  If not this means there are no tests covering this behavior and should be added.

Reply:
And you mention that `@each.id` will not work in Ember 2.12.x.  Does `[]` work in 2.11.x and earlier?  If not yet another reason this MUST be a MAJOR, and needs to be called out specifically in the changelog, which currently is not occurring.   Best practice is to open an issue where all of this can be captured and discussed and then this PR resolve it.

Reply:
It looks like only `addObject`,`removeAt`, and `setObjects` are used to set selected items (https://github.com/ciena-frost/ember-frost-list/blob/master/addon/utils/selection.js#L145-L149) I've never seen any set of selectedItems by looping through the items and changing the object itself (let alone it's `id`)

`.[]` works in 2.11.x it's common among all our packages and even used in the `ember-frost-list` to set the item states (https://github.com/ciena-frost/ember-frost-list/blob/master/addon/components/frost-list.js#L98). Additionally, if that dependent key is used to set our states it should be equally good to use it for our multi-select dependent

Reply:
@EWhite613 
> https://github.com/ciena-frost/ember-frost-list/blob/master/addon/utils/selection.js#L145-L149
This is only just one example usage of this open-source add-on.  Just because this specific example is not impacted by this change does not mean others are not.

> works in 2.11.x
Have you tested this in two different apps running these different versions of Ember?  The `ember-try` scenarios will but since there do not seem to be any tests around this behavior it isn't being leveraged.  Even if you have tested it extensively there still needs to be tests written around this.